operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java

```diff
@@ -286,7 +286,7 @@ public Labels withKubernetesPartOf(String instanceName) {
         while (i > 0) {
             char lastChar = instance.charAt(i - 1);
 
-            if (lastChar == '.' || lastChar == '-') {
+            if (lastChar == '.' || lastChar == '-' || lastChar == '_') {
```

Comment:
Considering how similar this method and the volume name one are, is it worth consolidating them down into a single 
```
makeValidKubernetesName
```
method (or names something similar)

Reply:
We should keep them separate since the rules are separate. For example `.` or `_` is legal in labale, just cannot be at the start or end. But in volume name, both `.` and `_` are illegal. I really just added the `_` here because I noticed it is missing.

Reply:
Makes sense to me, thanks for explaining!